packages/web-components/fast-ssr/src/fast-style/index.ts

```diff
@@ -0,0 +1,77 @@
+import { DOM, StyleTarget } from "@microsoft/fast-element";
+
+interface StyleCache {
+    [key: string]: CSSStyleSheet | string;
+}
+
+/**
+ * The FASTStyle web component that takes attributes:
+ * - css - a string version of the CSS to be applied to the parent web component, this can be omitted if the data-style-id has been used in the DOM as the cached value will be fetched
+ * - data-style-id - a dataset attribute used as an identifier for the CSS attr string
+ */
+export default class FASTStyle extends HTMLElement {
+    static cache: StyleCache = {};
+    static hashIdDataSetName: string = "data-style-id";
+
+    constructor() {
+        super();
+
+        const hashId: string = this.getAttribute(FASTStyle.hashIdDataSetName) as string;
```

Comment:
You should move all this code the the `connectedCallback` - https://html.spec.whatwg.org/multipage/custom-elements.html#custom-element-conformance recommends not inspecting attributes in the custom element constructor.